examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -0,0 +1,86 @@
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Threading.Tasks;
+
+namespace WebDemoWithEventHub
+{
+    public class EventHubService
+    {
+        private Settings _settings;
+
+        private EventHubConnection _eventHubConnection;
+
+        private EventProcessorClient _processorClient;
+
+        private IConfigRefresher _configRefresher;
+
+        public EventHubService(IConfiguration config, IConfigRefresher configRefresher)
+        {
+            _configRefresher = configRefresher;
+
+            _eventHubConnection = config.GetSection("WebDemo:Connection").Get<EventHubConnection>();
+            _settings = config.GetSection("WebDemo:Settings").Get<Settings>();
+
+            InitEventHubProcessor();
+        }
+
+        private void InitEventHubProcessor()
+        {
+            if (string.IsNullOrEmpty(_eventHubConnection.EventHubConnectionString))
+            {
+                throw new ArgumentNullException(nameof(_eventHubConnection.EventHubConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(_eventHubConnection.EventHubName))
+            {
+                throw new ArgumentNullException(nameof(_eventHubConnection.EventHubName));
+            }
+
+            if (string.IsNullOrEmpty(_eventHubConnection.StorageConnectionString))
+            {
+                throw new ArgumentNullException(nameof(_eventHubConnection.StorageConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(_eventHubConnection.StorageContainerName))
+            {
+                throw new ArgumentNullException(nameof(_eventHubConnection.StorageContainerName));
+            }
+
+            string consumerGroup = string.IsNullOrEmpty(_eventHubConnection.EventHubConsumerGroup)
+                ? EventHubConsumerClient.DefaultConsumerGroupName
+                : _eventHubConnection.EventHubConsumerGroup;
+
+            var storageClient = new BlobContainerClient(connectionString: _eventHubConnection.StorageConnectionString,
+                                                        blobContainerName: _eventHubConnection.StorageContainerName);
+
+            _processorClient = new EventProcessorClient(checkpointStore: storageClient,
+                                                       consumerGroup: consumerGroup,
+                                                       connectionString: _eventHubConnection.EventHubConnectionString,
+                                                       eventHubName: _eventHubConnection.EventHubName);
+
+            _processorClient.ProcessEventAsync += ProcessEventHandler;
+            _processorClient.ProcessErrorAsync += ProcessErrorHandler;
+            _processorClient.StartProcessing();
+        }
+
+        private Task ProcessEventHandler(ProcessEventArgs eventArgs)
```

Comment:
Can this function be async?

Reply:
It can be async, but it doesn't await on anything, right? This method could have a "void" type return statement but the handler method expects a Task.

---
In reply to: [427594593](https://github.com/Azure/AppConfiguration/pull/329#discussion_r427594593) [](ancestors = 427594593)